packages/sdk/src/swap/sdk.ts

```diff
@@ -159,46 +157,70 @@ export class SwapSDK {
     swapStatusRequest: SwapStatusRequest,
     options: RequestOptions = {},
   ): Promise<SwapStatusResponse> {
-    return ApiService.getStatus(this.baseUrl, swapStatusRequest, options);
+    return ApiService.getStatus(
+      this.options.backendUrl,
+      swapStatusRequest,
+      options,
+    );
   }
 
   async executeSwap(
     params: ExecuteSwapParams,
-    txOpts: TransactionOptions = {},
+    txOpts: TransactionOptions & { signer?: Signer } = {},
   ): Promise<TransactionHash> {
     const { srcChain, srcAsset, amount } = params;
 
+    const { signer: optsSigner, ...remainingTxOpts } = txOpts;
+    const signer = optsSigner ?? this.options.signer;
+    assert(signer, 'No signer provided');
+    assert(signer.provider, 'Signer has no provider');
+
+    const sourceChainId = getEvmChainId(params.srcChain, this.options.network);
+    const { chainId: signerChainId } = await signer.provider.getNetwork();
+    assert(
+      signerChainId === BigInt(sourceChainId ?? -1),
+      `Signer is connected to unexpected evm chain (expected: ${sourceChainId}, got: ${signerChainId})`,
+    );
+
     await this.validateSwapAmount(
       { chain: srcChain, asset: srcAsset },
       BigInt(amount),
     );
 
-    assert(this.signer, 'No signer provided');
     const receipt = await executeSwap(
       params,
       {
-        network: this.network,
-        signer: this.signer,
+        network: this.options.network,
+        signer,
       },
-      txOpts,
+      remainingTxOpts,
     );
     return receipt.hash as `0x${string}`;
   }
 
   async approveVault(
     params: Pick<TokenSwapParams, 'srcChain' | 'srcAsset' | 'amount'>,
-    txOpts: TransactionOptions = {},
+    txOpts: TransactionOptions & { signer?: Signer } = {},
   ): Promise<TransactionHash | null> {
-    if (!('srcAsset' in params)) return null;
-    assert(this.signer, 'No signer provided');
+    const { signer: optsSigner, ...remainingTxOpts } = txOpts;
+    const signer = optsSigner ?? this.options.signer;
+    assert(signer, 'No signer provided');
+    assert(signer.provider, 'Signer has no provider');
+
+    const sourceChainId = getEvmChainId(params.srcChain, this.options.network);
+    const { chainId: signerChainId } = await signer.provider.getNetwork();
+    assert(
+      signerChainId === BigInt(sourceChainId ?? -1),
+      `Signer is connected to unexpected evm chain (expected: ${sourceChainId}, got: ${signerChainId})`,
+    );
```

Comment:
Could we dedup this block? I see its identical to the one in `executeSwap` fn.

Reply:
extracted a `assertSignerIsConnectedToChain` util and moved the check down the call stack